libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryData.kt

```diff
@@ -0,0 +1,51 @@
+package com.mapbox.navigation.core.replay.history
+
+import com.mapbox.geojson.Point
+
+/**
+ * The translated history data. Every event in the list is assumed to
+ * increase in time. The replay starts at index 0 and moves to events.size over time.
+ */
+data class ReplayHistoryData(
+    val events: List<ReplayEventBase>
+)
+
+interface ReplayEventBase {
```

Comment:
Possibly, but it would change some things a bit. 
![Screen Shot 2020-04-02 at 2 24 36 PM](https://user-images.githubusercontent.com/3021882/78301293-b6bafb80-74ed-11ea-8aaa-5c6811fdce6d.png)

I chose interface to make it a requirement for all base classes to override the `event_timestamp`. The replay won't know when to emit events without it.

Reply:
> Possibly, but it would change some things a bit.

Yeah, I didn't mean explicitly the `interface` 😅 I meant the whole design around `ReplayEventBase` and yeah it needs some refactoring 👍 If it complicates things let's leave it as is for now. As we have a restricted hierarchy it smelled `sealed class` to me 😄

Reply:
haha yeah, the first pass was only to make eventTimestamp a required field.

to support custom replay events, went with an `abstract class`